vizro-core/src/vizro/managers/_data_manager.py

```diff
@@ -60,20 +68,34 @@ def _add_component(self, component_id: ComponentID, dataset_name: DatasetName):
             )
         self.__component_to_original[component_id] = dataset_name
 
-    def _get_component_data(self, component_id: ComponentID) -> pd.DataFrame:
-        """Returns the original data for `component_id`."""
-        if component_id not in self.__component_to_original:
-            raise KeyError(f"Component {component_id} does not exist. You need to call add_component first.")
-        dataset_name = self.__component_to_original[component_id]
-
+    @_cache.memoize()
+    def _get_original_data(self, dataset_name: DatasetName) -> pd.DataFrame:
+        """Returns the original data for `dataset_name`."""
         # Populate original data on first access only
+        logger.debug("get original data: %s", dataset_name)
+        logger.debug("loading...")
+        time.sleep(2.0)
         if dataset_name not in self.__original_data:
             self.__original_data[dataset_name] = self.__lazy_data[dataset_name]()
 
         # Return a copy so that the original data cannot be modified. This is not necessary if we are careful
         # to not do any inplace=True operations, but probably safest to leave it here.
         return self.__original_data[dataset_name].copy()
 
+    def _get_component_data(self, component_id: ComponentID) -> pd.DataFrame:
+        """Returns the original data for `component_id`."""
+        logger.debug("get_component_data: %s", component_id)
+        if component_id not in self.__component_to_original:
+            raise KeyError(f"Component {component_id} does not exist. You need to call add_component first.")
+        dataset_name = self.__component_to_original[component_id]
+
+        component_data = self._get_original_data(dataset_name)
+
+        # clean up original data if the cache type is not NullCache
+        self._clean_original_data()
```

Comment:
Why do we want this here and called again in the `Vizro.build`? 🤔

Reply:
Great catch! I now dropped the one in Vizro.build as it's duplicate. We need to put it here so it can prune the original data in two cases:
1. when the dashboard first build
2. when the cache is invalid, and a reload happens